File: esbmc.py/Testes02/validador_categoria/validador_comprimento.cpp
#include <cstdlib>

#define MAX_LEN 31

using namespace std;

bool validate_length(int len) {
    __ESBMC_assert(len >= 3 && len <= 30, "A categoria deve ter entre 3 e 30 caracteres.");
    return true;
}

int main() {
    int len;
    __ESBMC_assume(len >= 3 && len <= 30);

    validate_length(len);

    return 0;
}

File: esbmc.py/Testes02/validador_categoria/validador_comprimento.py
MAX_LEN: int = 31

def validate_length(length: int) -> bool:
    assert length >= 3 and length <= 30, \
        "A categoria deve ter entre 3 e 30 caracteres."
    return True


def main() -> None:
    category: str = input()        
    length: int = len(category)   

    assert length >= 3 and length <= 30

    validate_length(length)


main()
